22 feature improve demo UI

stefanfrench opened the pull request:
@Kostis-S-Z made some UI changes:

For the dataset app:
- Added a title and some instruction steps
- Improved the layout of index and previous/ next
- Changed it so that when you click save recording, you can then instantly record again instead of clicking on the 'x'
- Added queue() which should help experience

For the transcribe app:
- Added a title and some instruction steps

Kostis-S-Z commented on demo/transcribe_app.py:
```diff
@@ -34,6 +34,12 @@ def transcribe(pipe: Pipeline, audio: gr.Audio) -> str:
 def setup_gradio_demo():
     with gr.Blocks() as demo:
+        gr.Markdown(
+            "# 🗣️ Speech-to-Text Transcription\n"
+            "1️. Select a model** and a **language** from the dropdowns. \n"
+            "2️. Load the model** by clicking the **Load model** button. \n"
+            "3️. Record a message** and click **Transcribe** to see the transcription. \n"
```
```suggestion
            """ # 🗣️ Speech-to-Text Transcription
            ### 1. Select a model and a language from the dropdowns.
            ### 2. Load the model by clicking the Load model button.
            ### 3. Record a message and click Transcribe to see the transcription.
            """
```
Same thing as the dataset recorder app

stefanfrench commented:
@Kostis-S-Z - thanks for reviewing. I removed the queue function as its not doing anything useful as far as i can tell. I added it originally as i was experimenting on trying to add a loading spinner for when the dataset loads, but it didn't work as expected.

Let me know and I will merge this in

Kostis-S-Z commented:
Sounds good! Just run `ruff format` to fix the linting error and we merge it!
